lib/core/src/Cardano/Wallet/DB/Store/Wallets/Store.hs

```diff
@@ -0,0 +1,122 @@
+    ( mkStoreMetaTransactions )
+import Cardano.Wallet.DB.Store.Transactions.Model
+    ( DeltaTxHistory (..), TxHistoryF (..), mkTxHistory )
+import Cardano.Wallet.DB.Store.Transactions.Store
+    ( mkStoreTransactions )
+import Cardano.Wallet.DB.Store.Wallets.Model
+    ( DeltaTxWalletsHistory (..), walletsLinkedTransactions )
+import Control.Applicative
+    ( liftA2 )
+import Control.Exception
+    ( SomeException )
+import Control.Monad
+    ( forM, forM_ )
+import Control.Monad.Except
+    ( ExceptT (ExceptT), runExceptT )
+import Data.DBVar
+    ( Store (..) )
+import Data.DeltaMap
+    ( DeltaMap (..) )
+import Data.Generics.Internal.VL
+    ( view )
+import Data.List
+    ( nub )
+import Database.Persist.Sql
+    ( SqlPersistT, deleteWhere, entityVal, selectList, (==.) )
+
+import qualified Cardano.Wallet.DB.Store.Meta.Model as TxMetaStore
+import qualified Cardano.Wallet.DB.Store.Transactions.Model as TxStore
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Data.Map.Strict as Map
+
+-- | Store for 'WalletsMeta' of multiple different wallets.
+mkStoreWalletsMeta :: Store
+        (SqlPersistT IO)
+        (DeltaMap W.WalletId DeltaTxMetaHistory)
+mkStoreWalletsMeta =
+    Store
+    { loadS = load
+    , writeS = write
+    , updateS = update
+    }
+  where
+    write reset = forM_ (Map.assocs reset) $ \(wid,metas)
+        -> writeS (mkStoreMetaTransactions wid) metas
+    update _ (Insert wid metas) =
+        writeS (mkStoreMetaTransactions wid) metas
+    update _ (Delete wid) = do
+        deleteWhere [TxMetaWalletId ==. wid ]
+    update _ (Adjust wid da) =
+        updateS (mkStoreMetaTransactions wid) undefined da
+    load :: SqlPersistT
+            IO
+            (Either SomeException (Map.Map W.WalletId TxMetaHistory))
+    load = do
+        wids <- nub . fmap (view #txMetaWalletId . entityVal)
+            <$> selectList @TxMeta [] []
+        runExceptT $ do
+            xs <- forM wids $ ExceptT . loadS . mkStoreMetaTransactions
+            pure $ Map.fromList $ zip wids xs
+
+mkStoreTxWalletsHistory
+    :: Store (SqlPersistT IO) DeltaTxWalletsHistory
+mkStoreTxWalletsHistory =
+    Store
+    { loadS =
+            liftA2 (,) <$> loadS mkStoreTransactions <*> loadS mkStoreWalletsMeta
+    , writeS = \(txHistory,txMetaHistory) -> do
+            writeS mkStoreTransactions txHistory
+            writeS mkStoreWalletsMeta txMetaHistory
+    , updateS = \(txh@(TxHistoryF mtxh),mtxmh) -> \case
+            ExpandTxWalletsHistory wid cs -> do
+                updateS mkStoreTransactions txh
+                    $ TxStore.Append
+                    $ mkTxHistory
+                    $ fst <$> cs
+                updateS mkStoreWalletsMeta mtxmh
+                    $ Adjust wid
+                    $ TxMetaStore.Expand
+                    $ mkTxMetaHistory wid cs
+            ChangeTxMetaWalletsHistory wid change
+                -> updateS mkStoreWalletsMeta mtxmh
+                $ Adjust wid
```

Comment:
Ok, here I think we need a comment on the "thing where the database does not distinguish between an empty table and no wallet" — Hm. 🤔 Or we make a speaking name `removeEmptyWallets` and use it in both place — In one place, the definition is `removeEmptyWallets = Map.filter (not . null . relations)` while here, the definition is `removeEmptyWallets = id` or another no-op.